source/simulator/melt.cc

```diff
@@ -1221,10 +1231,83 @@ namespace aspect
       std_cxx11::shared_ptr<MaterialModel::AdditionalMaterialOutputs<dim> >
       (new MaterialModel::MeltOutputs<dim> (n_points, n_comp)));
   }
-}
 
 
 
+  template <int dim>
+  void
+  MeltHandler<dim>::set_melt_assemblers(const SimulatorAccess<dim> &simulator_access,
+                                        internal::Assembly::AssemblerLists<dim> &assemblers,
+                                        std::vector<dealii::std_cxx11::shared_ptr<
+                                        internal::Assembly::Assemblers::AssemblerBase<dim> > > &assembler_objects)
+  {
+    if (!simulator_access.get_parameters().include_melt_transport)
+      return;
+
+    std_cxx11::shared_ptr<Assemblers::MeltEquations<dim> > melt_equation_assembler
+    (new Assemblers::MeltEquations<dim>());
+    assembler_objects.push_back (melt_equation_assembler);
+
+    assemblers.local_assemble_stokes_preconditioner.disconnect_all_slots();
```

Comment:
how does this interact with the free surface assembler for example? I think this is added inside the constructor of ``FreeSurfaceHandler``. If you run with free surface and melt, this will break.